File: 2021/vika2/frosting.py
def run(n,p,q):
    dic = {
        0 : 0,
        1 : 0,
        2 : 0
    }
    """
    p = [1,1,1]
    q = [1,1,1]
    """
    for i in range(len(p)):
        for j in range(len(q)):
            dic[(i+j) % 3] += p[i]*q[j]

    #print(' '.join(map(str,dic.values())))
    print(dic[1],dic[2],dic[0])

n = int(input())
p = input().split()
p = [int(i) for i in p]

q = input().split()
q = [int(i) for i in q]

a = 3 <= n <= 100000
b = 1 <= p[0] and 1 <= q[0]
c = p[n-1] <= 10000 and q[n-1] <= 10000

if a and b and c:
    run(n,p,q)

File: 2021/vika2/frosting.cpp
#include<iostream>
using namespace std;

void print_array(int arr[], int size) {
    for (int i = 0; i < size; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
}

void run(int n, int p[], int q[]) {
    int dic[] = {0,0,0};

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            dic[(i+j) % 3] += p[i] * q[j];
        }
    }

    //print_array(dic,3);
    cout << dic[1] <<  " " << dic[2] << " " << dic[0] << endl;
}

int main() {
    int n;
    cin >> n;

    int p[n];
    int q[n];

    for (int i = 0; i < n; i++) {
        cin >> p[i];
    }
    for (int i = 0; i < n; i++) {
        cin >> q[i];
    }
/*
    cout << n << endl;
    print_array(p,n);
    print_array(q,n);
*/
    run(n,p,q);
    return 0;
}
